public/examples/latency/latencyApp.js

```diff
@@ -231,72 +231,135 @@
+          if(result.results.length===0) {
+            //set test value to 0
+            option.series[0].data[0].value = 0;
+            //updat test status to complete
+            option.series[0].data[0].name = 'Test Failed';
+            //set accessiblity aria-disabled state.
+            //This will also effect the visual look by corresponding css
+            startTestButton.setAttribute('aria-disabled', false);
+            //update button text to communicate current state of test as In Progress
+            startTestButton.innerHTML = 'Start Test';
+            //enable start button
+            startTestButton.disabled = false;
+            //hide current test value in chart
+            option.series[0].detail.show = false;
+            //update gauge
+            myChart.setOption(option, true);
+          }else{
+            result = result.results.sort(function (a, b) {
+              return +a.time - +b.time;
+            });
+            updateValue(currentTest, result[0].time + ' ms');
+          }
+          if (version === 'IPv6') {
+            latencyTest('IPv4');
+            return;
+          }
+          else{
+            void (!(version === 'IPv6') && setTimeout(function () {
+              //update button text to communicate current state of test as In Progress
+              startTestButton.innerHTML = 'Start Test';
+              option.series[0].data[0].value = 0;
+              option.series[0].data[0].name = 'Test Complete';
+              //set accessiblity aria-disabled state.
+              //This will also effect the visual look by corresponding css
+              startTestButton.setAttribute('aria-disabled', false);
+              startTestButton.disabled = false;
+              option.series[0].detail.show = false;
+              myChart.setOption(option, true);
+
+            }, 500));
+          }
         }
 
         function latencyHttpOnTimeout(result) {
-            if (version === 'IPv6') {
-                testPlan.hasIPv6 = false;
-                latencyTest('IPv4');
-                return;
-            }
-                //set test value to 0
-                option.series[0].data[0].value = 0;
-                //updat test status to complete
-                option.series[0].data[0].name = 'Test Failed';
-                //set accessiblity aria-disabled state. 
-                //This will also effect the visual look by corresponding css
-                startTestButton.setAttribute('aria-disabled', false);
-               //update button text to communicate current state of test as In Progress
-                startTestButton.innerHTML = 'Start Test';
-                //enable start button
-                startTestButton.disabled = false;
-                //hide current test value in chart 
-                option.series[0].detail.show = false;
-                //update gauge
-                myChart.setOption(option, true);
+          if(result.results.length===0) {
```

Comment:
can be result && result.results && result.results.length === 0 or to make it solid.